src/csharp/Grpc.Core.Tests/MockServiceHelper.cs

```diff
@@ -184,6 +184,10 @@ public ServerServiceDefinition ServiceDefinition
             {
                 return this.serviceDefinition;
             }
+            set
```

Comment:
Just turn this into an automatically implemented property? (Ditto for Host potentially, but that's not being changed in this PR, so it may want to be done separately.)

Reply:
I considered this, but all (or almost all?) properties in gRPC C# code base seem to be not automatically generated. I wonder if it is intentional for compatibility purposes with older compilers. That's why I kept consistent with the approach in the codebase.

Reply:
ad auto-properties: there's no particular reason why we are not using them, but you're right the most times I didn't use them so for now it might makes sense to stay consistent.

I think autoproperties have been in C# since early version, so compiler compatibility is not a concern here. But there are other language constructs (from C# 6) that we intentionally didn't use because until about a year ago, we needed the code to compile on older mono and Visual Studio. Now when we switched to the new dotnet projects, we could start using C# 6 in our code (the compiled code will still be compatible with older .NET and the generated code needs to still compile with older .NET) - so feel free to use C# 6 if you want to. Sometime in the future, we might refactor portions of our existing code and introduce C# 6 constructs where appropriate.